Stale `ANTHROPIC_API_KEY` in host env silently shadows working subscription auth

## Summary

When both `ANTHROPIC_API_KEY` is set in the host shell **and** `~/.claude/.credentials.json` exists, benchflow silently prefers the env var — even if the env var is stale/invalid. The failure surfaces deep in the ACP stack as `ACP error -32603: Invalid API key`, with no hint that working subscription auth was available and bypassed.

## Repro

On a host with:
- A stale/invalid `ANTHROPIC_API_KEY` exported (e.g. from a previous dev session)
- A valid `~/.claude/.credentials.json` from `claude login`

Run the live smoke test:

```bash
uv run pytest -m live tests/test_smoke.py
```

Result: `smoke_prereqs` green-lights the run (it only checks *presence*, not validity — see `test_smoke.py:_smoke_skip_reason`), then `SDK.run()` ships the invalid env var into the container and the ACP agent rejects it:

```
E   benchflow.acp.client.ACPError: ACP error -32603: Internal error: Invalid API key · Fix external API key
```

Workaround: `env -u ANTHROPIC_API_KEY uv run pytest -m live tests/test_smoke.py`

## Root cause

`src/benchflow/_agent_env.py:133-160` (`resolve_agent_env`) calls `auto_inherit_env(agent_env)` first, which pulls `ANTHROPIC_API_KEY` from the process environment. Then at line 148:

```python
if required_key and required_key not in agent_env:
    if check_subscription_auth(agent, required_key):
        agent_env["_BENCHFLOW_SUBSCRIPTION_AUTH"] = "1"
        ...
```

Because `required_key` is already in `agent_env` (inherited from the host), the `check_subscription_auth` branch is skipped. The stale env var wins silently.

## Suggested fixes (any one of these would help)

1. **Warn when both are present.** When `auto_inherit_env` pulls in `ANTHROPIC_API_KEY` *and* a subscription-auth credentials file exists for the agent, log a WARNING so the user knows the env var is taking precedence.
2. **Validate at the edge.** In `smoke_prereqs` (or `resolve_agent_env`), do a lightweight shape check on `ANTHROPIC_API_KEY` (e.g. starts with `sk-ant-`) and fail/warn early if it's malformed.
3. **Retry with subscription auth on `Invalid API key`.** Catch the specific ACP error and retry once with `_BENCHFLOW_SUBSCRIPTION_AUTH=1` if the credentials file is available. (Riskier — silently papers over misconfiguration.)

Option 1 is the cheapest and most honest. A single log line at `WARNING` level from `resolve_agent_env` would have saved ~4 minutes of debugging in my session.

## Context

Hit this while running the new e2e smoke test on `main @ 242f82c` in a DinD devcontainer. Detail in this Discord thread (internal): https://discord.com/channels/.../1492980523172102195

--- a/src/benchflow/_agent_env.py
+++ b/src/benchflow/_agent_env.py
@@ -158,16 +158,19 @@ def resolve_agent_env(
                     f"Export it, pass via agent_env, or log in with the "
                     f"agent CLI (e.g. claude login, codex --login)."
                 )
-        elif required_key and required_key in agent_env:
-            if check_subscription_auth(agent, required_key):
-                logger.warning(
-                    "%s is set (possibly inherited from host env) AND "
-                    "subscription auth credentials exist — the env var takes "
-                    "precedence. If the key is stale, unset it: "
-                    "env -u %s <command>",
-                    required_key,
-                    required_key,
-                )
+        elif (
+            required_key
+            and required_key in agent_env
+            and check_subscription_auth(agent, required_key)
+        ):
+            logger.warning(
+                "%s is set (possibly inherited from host env) AND "
+                "subscription auth credentials exist — the env var takes "
+                "precedence. If the key is stale, unset it: "
+                "env -u %s <command>",
+                required_key,
+                required_key,
+            )
     else:
         # No model specified — still check subscription auth for required env vars
         agent_cfg = AGENTS.get(agent)

--- a/src/benchflow/_agent_env.py
+++ b/src/benchflow/_agent_env.py
@@ -158,6 +158,16 @@ def resolve_agent_env(
                     f"Export it, pass via agent_env, or log in with the "
                     f"agent CLI (e.g. claude login, codex --login)."
                 )
+        elif required_key and required_key in agent_env:
+            if check_subscription_auth(agent, required_key):
+                logger.warning(
+                    "%s is set (possibly inherited from host env) AND "
+                    "subscription auth credentials exist — the env var takes "
+                    "precedence. If the key is stale, unset it: "
+                    "env -u %s <command>",
+                    required_key,
+                    required_key,
+                )
     else:
         # No model specified — still check subscription auth for required env vars
         agent_cfg = AGENTS.get(agent)

--- a/src/benchflow/sdk.py
+++ b/src/benchflow/sdk.py
@@ -120,7 +120,7 @@ from benchflow._trajectory import (
     _capture_session_trajectory,
     _scrape_agent_trajectory,
 )
-from benchflow.acp.client import ACPClient
+from benchflow.acp.client import ACPClient, ACPError
 from benchflow.agents.registry import AGENT_LAUNCH
 from benchflow.models import RunResult, TrajectorySource
 
@@ -579,6 +579,23 @@ class SDK:
         except ConnectionError as e:
             error = str(e)
             logger.error(f"Agent connection lost: {error}")
+        except ACPError as e:
+            if "Invalid API key" in e.message:
+                from benchflow._agent_env import check_subscription_auth
+                from benchflow.agents.registry import infer_env_key_for_model
+
+                key = infer_env_key_for_model(model) if model else None
+                if key and check_subscription_auth(agent, key):
+                    error = (
+                        f"{key} was rejected as invalid. "
+                        f"Subscription auth credentials exist — unset the env var "
+                        f"to use them: env -u {key} <command>"
+                    )
+                else:
+                    error = str(e)
+            else:
+                error = str(e)
+            logger.error(error)
         except Exception as e:
             error = str(e)
             logger.error("Run failed", exc_info=True)

--- a/tests/test_smoke.py
+++ b/tests/test_smoke.py
@@ -105,6 +105,12 @@ def smoke_jobs_dir(tmp_path: Path) -> Iterator[Path]:
     may not be removable by our (non-root) test user.
     """
     if _detect_dind_mount() is None:
+        if Path("/.dockerenv").exists():
+            pytest.fail(
+                "Running inside DinD (/.dockerenv present) but cwd is not under any "
+                "container bind mount. Move your checkout under a bind-mounted path "
+                "(e.g. /workspace) so host docker can see it."
+            )
         yield tmp_path
         return
 

